internal/command/test.go

```diff
@@ -130,21 +130,37 @@ func (c *TestCommand) Run(rawArgs []string) int {
 		return 1
 	}
 
-	// Per file, ensure backends aren't reused
-	var duplicateBackendDiags tfdiags.Diagnostics
+	// Per file, ensure backends:
+	// * aren't reused
+	// * are valid types
+	var backendDiags tfdiags.Diagnostics
 	for _, tf := range config.Module.Tests {
 		bucketHashes := make(map[int]string)
 		// Use an ordered list of backends, so that errors are raised by 2nd+ time
 		// that a backend config is used in a file.
 		for _, bc := range orderBackendsByDeclarationLine(tf.BackendConfigs) {
 			f := backendInit.Backend(bc.Backend.Type)
+			if f == nil {
+				detail := fmt.Sprintf("There is no backend type named %q.", bc.Backend.Type)
+				if msg, removed := backendInit.RemovedBackends[bc.Backend.Type]; removed {
```

Comment:
This validation needs to be here because it also depends on internal/backend/init, which is hard to import elsewhere